place_tracker/lib/place_details.dart

```diff
@@ -0,0 +1,215 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+import 'place.dart';
+
+class PlaceDetails extends StatefulWidget {
```

Comment:
Classes like this usually do not mutate the state (place) that's passed to them. They just notify the parent widget that the user wants a change So:

```
class PlaceDetails extends StatefulWidget {
  const PlaceDetails({
    Key key,
    @required this.place,
    @required this.onChanged,
  }) : /* TBD asserts */ super(key: key);

  final Place place;
  final ValueChanged<Place> onChanged;

  @override
  PlaceDetailsState createState() => PlaceDetailsState();
}
```

The Stateful parent provides the `onChanged` callback and rebuilds its PlaceDetails child with the new place value.

The state class would not have a _place field, it would just refer to `widget.place`.

Reply:
I implemented this with a slight modification. I only want to notify the PlaceMap that the place has changed if the user presses the save button. Otherwise, the user would not have the ability to revert accidental edits (which they can do by just pressing back). So I made a local _place within PlaceDetailsState and that is the Place that the PlaceDetails page displays. Upon pressing save, this local place is passed back to the PlaceMap via the onChanged method of PlaceDetails.